### ISSUE TITLE
use system colors for the shell

### ISSUE THREAD

**Issue by flying-sheep** [glennw/servo-shell#6]
UI should use [system colors](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#System_Colors) to integrate

**Comment by jdm** [comment]
@flying-sheep Want to file a corresponding issue in rust-cssparser to figure out how to support nonstandard properties like this? The idea of adding platform-specific code to https://github.com/servo/rust-cssparser/blob/master/src/color.rs#L99 to figure these values out isn't particularly appealing :)

**Comment by flying-sheep** [comment]
nonstandard? [where did you get that idea from](http://www.w3.org/TR/css3-color/#css-system)?

will do. and yeah, that doesn’t sound too clean given that it needs to be resolved at runtime (OS color themes can change at any time and differ between applications)

but there already is the special value `Color::CurrentColor`, so we’d just need to add 28 more special values.

and yeah, as said in the standard, we could use `apperance` instead, but i guess that’s more work… just think about what `apparance: tab` would imply instead of a simple `color: WindowText; background: Window; border: 1px solid WindowFrame`. besides, the [current version of CSS3-UI](http://www.w3.org/TR/css3-ui/) has no mention of `apperance`, so the deprecation notice is pretty unhinged (deprecated in favor of something that was decided against?)

**Comment by flying-sheep** [comment]
fyi: the issue exists, and is the only currently open one: servo/rust-cssparser#2

**Comment by jdm** [comment]
Huh, TIL. For what it's worth, the TR/ specs should be ignored in favour of http://dev.w3.org/csswg/: http://dev.w3.org/csswg/css-color-3/#css-system, http://dev.w3.org/csswg/css-color/#system-colors, and http://dev.w3.org/csswg/css-ui-4/#appearance-switching

**Comment by flying-sheep** [comment]
yeah, but as i said there: `appearance` has only the “button” value, so it’s useless, and system colors are the closest to get a semblance of native appearance.

**Comment by flying-sheep** [comment]
That issue has been closed, system colors are back in the standard (not all of them, many individual ones have been deprecated):

- https://www.w3.org/TR/css-color-4/#css-system-colors
- https://www.w3.org/TR/css-color-4/#deprecated-system-colors
